frontend/src/utils/StudentUtils.ts

```diff
@@ -0,0 +1,57 @@
+import update from "immutability-helper";
+
+import { QuestionComponentRequest } from "../APIClients/types/TestClientTypes";
+import { Answers } from "../types/AnswerTypes";
+import { ResponseElementType } from "../types/QuestionTypes";
+
+export const getAnswerElements = (
+  question: QuestionComponentRequest[],
+): QuestionComponentRequest[] => {
+  return question.filter(
+    (questionElement) => questionElement.type in ResponseElementType,
+  );
+};
+
+export const getAnswerValues = (
+  currentQuestionIndex: number,
+  answerElementIndex: number,
+  answers: Answers[],
+): number[] => {
+  const answer = answers.find((a) => a.index === currentQuestionIndex);
+  const answerElement = answer?.elements[answerElementIndex];
+  return answerElement?.elementAnswers ?? [];
+};
+
+export const initializeAnswers = (
+  questions: QuestionComponentRequest[][],
+): Answers[] => {
+  return questions.map((question, index) => ({
+    index,
+    elements: getAnswerElements(question).map((_, elementIndex) => ({
+      index: elementIndex,
+      elementAnswers: [],
+    })),
+  }));
+};
+
+export const updatedAnswer = (
```

Comment:
Any reason not to use `updateAnswer`? Or something like `replaceAnswer`

Reply:
in cases where `updatedAnswer` is called, it's called by an `updateAnswer` function. Whereas this function contains the logic to construct a new Answers array (hence, `updatedAnswers`), the calling `updateAnswer` function calls the setter, `setAnswer` which actually updates the `answer` field in the `StudentContext`.

let me know if that doesn't make sense? i'm also happy to change it to `replaceAnswer` or something else, if you those are more precise. (:

Reply:
In general I like to start functions with a verb since it makes it clear what it's doing (e.g. `getUpdatedAnswers` or `updateAnswerArray` or `replaceAnswer[InAnswerArray]`). It's a really minor thing though, you can ignore it if you want

Reply:
makes sense to me. updated!